src/provider/model/AppDirectory.ts

```diff
@@ -0,0 +1,23 @@
+import {Application} from "../../client/directory";
+import {injectable} from 'inversify';
+
+@injectable()
+export class AppDirectory {
+    private _directory: Application[] = require('../../../res/provider/app-directory.json');
+
+    public async getAppByName(name: string): Promise<Application|null> {
+        return this._directory.find((app: Application) => {
+            return app.name === name;
+        }) || null;
+    }
+
+    public async getAppsByIntent(intentType: string): Promise<Application[]> {
```

Comment:
Is there any sense in also having a `getAppsByContext` for when we implement `findIntentsByContext` or is that likely to be handled differently?

Reply:
Yep - all of this will be built out further down the road. Still lots to do :-)